src/utils/get-size-by-breakpoint/get-size-by-breakpoint.ts

```diff
@@ -0,0 +1,35 @@
+import type { Size } from '~/types'
+
+import { type Breakpoint, breakpoints } from '~/styles'
+
+type GetSizeByBreakpointProps = {
+  size?: Size
+  breakpoint: Breakpoint
+}
+
+/**
+ * Gets the size of an element based on the current breakpoint and the provided size object.
+ *
+ * @param {GetSizeByBreakpointProps} props - The size and breakpoint information.
+ * @returns The size for the given breakpoint or a fallback value.
+ */
+export const getSizeByBreakpoint = ({
+  size,
+  breakpoint,
+}: GetSizeByBreakpointProps) => {
+  if (typeof size === 'number' || size === undefined) {
+    return size
+  }
+
+  const breakpointsOrder = Object.keys(breakpoints) as Breakpoint[]
+  const breakpointIndex = breakpointsOrder.indexOf(breakpoint)
+
+  if (breakpointIndex === -1) return undefined
+
+  const closestSize = breakpointsOrder
+    .slice(0, breakpointIndex + 1)
+    .reverse()
+    .find(bp => size[bp] !== undefined)
+
```

Comment:
Suggestion: Reversing and slicing the array could be optimized by iterating backwards through the array instead of reversing it. This would save on memory and performance for large breakpoint lists.